### ISSUE TITLE
Remove nodeInfo endpoint from kubelet

### ISSUE THREAD

**Issue by ddysher** [GoogleCloudPlatform/kubernetes#6073]
/nodeInfo endpoint is added for nodecontroller to pull node info from kubelet.  We need to evaluate if it's still necessary to keep it given this #6005

NodeInfo is also confusing in our api object

```go
// NodeSystemInfo is a set of ids/uuids to uniquely identify the node.
type NodeSystemInfo struct {
...
}

// NodeStatus is information about the current status of a node.
type NodeStatus struct {
	NodeInfo NodeSystemInfo `json:"nodeInfo,omitempty"`
}

// NodeInfo is the information collected on the node.
type NodeInfo struct {
	NodeSystemInfo `json:",inline,omitempty"`
}
```

@davidopp @bgrant0607 @gmarek

**Comment by bgrant0607** [comment]
The plan was to get rid of /nodeinfo. 

I was also planning to rename type NodeSystemInfo to type NodeInfo, but maybe we should rename the field to NodeSystemInfo instead, since that seems slightly more informative to me.

As part of #2098, I think it would be useful for Kubelet to export its current value of Node - either a copy of the last state sent to apiserver or the most recently updated state that will be next sent to apiserver (or both).

**Comment by davidopp** [comment]
Kubelet is also using the /nodeInfo endpoint:
https://github.com/GoogleCloudPlatform/kubernetes/blob/984bc8d5f6fd235229e8d2b0af029a5d2bf585c0/pkg/client/kubelet.go#L135

**Comment by gmarek** [comment]
IIUC this means we can't get rid of it easily. Do you think it's still worth doing?

**Comment by ddysher** [comment]
@davidopp  If I recall correctly, /nodeinfo endpoint is only used by nodecontroller now, the link is where kubelet expose the endpoint

@gmarek I don't expect much difficulty to remove it.  It's added here #5030 for node controller to fetch node info.  Once you remove the sync_status=true, it should be easy.

I'm not following the thread #2098, but @bgrant0607 are you suggesting we keep exposing node info even if node controller stops probing?  I'm fine with that, but if so, I want to clean up the types I mentioned above, it's really confusing.

**Comment by bgrant0607** [comment]
No, I was suggesting that the Kubelet expose the same Node API as the apiserver, just for that Kubelet.

**Comment by bgrant0607** [comment]
@davidopp @gmarek pkg/client/kubelet.go is the Kubelet client. Kubelet isn't using that, it's exporting that for the master.

**Comment by ddysher** [comment]
ok.  @gmarek I'm assigning this to myself for load balance the work.

**Comment by davidopp** [comment]
Yeah, sorry, what I saw was the definition, not another use. Looks like we can remove this.

**Comment by bgrant0607** [comment]
cc @nikhiljindal re #5475

**Comment by ddysher** [comment]
Closed by  #6702
